src/EventListener/AddToSearchIndexListener.php

```diff
@@ -40,9 +40,9 @@ public function __construct(ContaoFrameworkInterface $framework)
     /**
      * Forwards the request to the Frontend class if there is a page object.
      *
-     * @param PostResponseEvent $event
+     * @param FilterResponseEvent $event
      */
-    public function onKernelTerminate(PostResponseEvent $event)
+    public function onKernelResponse(FilterResponseEvent $event)
```

Comment:
The workflow was like the following:
1. `kernel.request` (master request) => `LegacyRequestValueSynchronizingListener` sets request.
2. `kernel.request` (sub request) => `LegacyRequestValueSynchronizingListener` sets request.
3. `kernel.response` => `LegacyRequestValueSynchronizingListener` sets parent request (master request).
4. `kernel.response` => `LegacyRequestValueSynchronizingListener` sets parent request (no request anymore as no parent).
5. `kernel.terminate` => `AddToSearchIndexListener` tries to access `Environment` which has no request anymore. :bomb: 

Now it is:
1. `kernel.request` (master request) => `LegacyRequestValueSynchronizingListener` sets request.
2. `kernel.request` (sub request) => `LegacyRequestValueSynchronizingListener` sets request.
3. `kernel.response` => `LegacyRequestValueSynchronizingListener` sets parent request (master request).
4. `kernel.response` => `AddToSearchIndexListener` accesses `Environment` which has the master request.
5. `kernel.response` => `LegacyRequestValueSynchronizingListener` sets parent request (no request anymore as no parent).

This is of course an intermediate change because as soon as I finish my work on the crawler, the whole `AddToSearchIndexListener` will get removed.